File: frontend/src/types/project.ts
export interface UserAvatar {
    email: string
    photo_url: string | null
}

export interface Project {
    project_id: number
    name: string
    domain: string
    users: UserAvatar[]
    created_at: string
}

export interface ProjectsResponse {
    success: boolean
    data: Project[]
}

export interface ProjectResponse {
    success: boolean
    data: Project
}

export interface CreateProjectPayload {
    name: string
    domain: string
}

export interface AddUserPayload {
    email: string
    project_id: number
}
export interface DeleteUserPayload {
    email: string
    project_id: number
}
export interface EditProjectPayload {
    project_id: number
    name: string
    domain: string
}

File: backend/type/payload/project.go
package payload

import "time"

type CreateProject struct {
	Name   *string `json:"name"`
	Domain *string `json:"domain"`
}
type EditProject struct {
	Name      *string `json:"name"`
	Domain    *string `json:"domain"`
	ProjectId *uint64 `json:"project_id"`
}

type User struct {
	Email    *string `json:"email"`
	PhotoUrl *string `json:"photo_url"`
}

type Project struct {
	ProjectId uint64    `json:"project_id"`
	Name      string    `json:"name"`
	Domain    string    `json:"domain"`
	Users     []User    `json:"users"`
	CreatedAt time.Time `json:"created_at"`
}
